server/e2e_test_command.go

```diff
@@ -249,3 +232,22 @@ func (s *Server) getBranchAndSHAWithSameName(ctx context.Context, owner string,
 	}
 	return ghBranch.GetName(), ghBranch.GetCommit().GetSHA(), nil
 }
+
+func (s *Server) waitForImageStatus(pr *model.PullRequest) (bool, error) {
+	ghStatusContext := ""
+	switch pr.RepoName {
+	case s.Config.E2EWebappReponame:
+		ghStatusContext = s.Config.E2EWebappStatusContext
+	case s.Config.E2EServerReponame:
+		ghStatusContext = s.Config.E2EServerStatusContext
+	}
+
+	ctx2, cancel := context.WithDeadline(context.Background(), time.Now().Add(s.Config.E2ETestDeadline))
```

Comment:
The value of `s.Config.E2ETestDeadline)` at the deployment is 7200 which I believe is 2 minutes. Isn't it very long duration?

Reply:
I wanted to give it enough time to be able to wait for building image. I am fine with something shorter though. What do you think would be a good time? 5 mins polling every 30 secs?

Reply:
Ah, I'm not correct, 7200 is 2 hours. Yup, 5 mins polling every 30 secs sounds good to me. Then, let the user read the message that the image is not yet available, and then do the retry manually as they want to.